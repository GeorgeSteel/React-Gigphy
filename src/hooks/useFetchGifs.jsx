import { useState, useEffect } from 'react'
import { getGifs } from '../helpers/getGifs'

export default function useFetchGifs(category) {
  const [state, setState] = useState({
    data: [],
    loading: true,
  })

  useEffect(() => {
    const fetchGifs = async () => {
      const imgs = await getGifs(category)
      setState({
        data: imgs,
        loading: false,
      })
    }

    fetchGifs()
  }, [category])

  return state
}
