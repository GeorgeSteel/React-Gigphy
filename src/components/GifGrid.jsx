import React from 'react'
import PropTypes from 'prop-types'

import useFetchGifs from '../hooks/useFetchGifs'

import GifGridItem from './GifGridItem'

function GifGrid({ category }) {
  const { data, loading } = useFetchGifs(category)

  return (
    <>
      <h3>{category}</h3>

      {loading && <p>Loading...</p>}

      <div className="card-grid">
        {data.map((gif) => (
          <GifGridItem key={gif.id} {...gif} />
        ))}
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}

export default GifGrid
