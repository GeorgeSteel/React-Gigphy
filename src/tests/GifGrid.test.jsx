import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

import GifGrid from '../components/GifGrid'
import useFetchGifs from '../hooks/useFetchGifs'

jest.mock('../hooks/useFetchGifs')

describe('GifGrid Component Tests', () => {
  const category = 'Saitama'

  test('should show the component without issues', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    })

    const wrapper = shallow(<GifGrid category={category} />)

    expect(wrapper).toMatchSnapshot()
  })

  test('should show items when the gifs are being loaded by using useFetchGifs', () => {
    const gifs = [
      { id: '123', url: 'https://localhost/any/thing.gif', title: 'anything' },
      {
        id: 'ABC',
        url: 'https://localhost/other/thing.gif',
        title: 'otherthing',
      },
    ]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    })

    const wrapper = shallow(<GifGrid category={category} />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('p').exists()).toBe(false)
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  })
})
