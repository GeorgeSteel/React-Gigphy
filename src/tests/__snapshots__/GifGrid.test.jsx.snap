// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Component Tests should show items when the gifs are being loaded by using useFetchGifs 1`] = `
<Fragment>
  <h3>
    Saitama
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="123"
      key="123"
      title="anything"
      url="https://localhost/any/thing.gif"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="otherthing"
      url="https://localhost/other/thing.gif"
    />
  </div>
</Fragment>
`;

exports[`GifGrid Component Tests should show the component without issues 1`] = `
<Fragment>
  <h3>
    Saitama
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
